### PR TITLE
[test] Trigger failures via USR1/USR2

### PR DESCRIPTION
**Pull request by Arpafaucon**
Add signal handlers to trigger special conditions in `tracy-test`. This proves useful when working on the `capture` utility to check edge conditions.

### USR1: abort the program

example of call & output

maim_2024-07-17T17-01-39+02-00

in the profiler: 

maim_2024-07-17T16-38-32+02-00

### USR2: send buggy `TracyFree` event, to trigger an instrumentation failure

```
❯ timeout --signal USR2 0.4 ./tracy-test
Triggering instrumentation failure
terminate called without an active exception
timeout: the monitored command dumped core
```

maim_2024-07-17T17-01-09+02-00

### CODE REVIEW

File: test/test.cpp [hunk 1]
```diff
@@ -1,4 +1,6 @@
+#include <atomic>
 #include <chrono>
+#include <csignal>
```
Comment (wolfpld) [inline test/test.cpp:3]: Use `signal.h`.

Comment (Arpafaucon) [reply test/test.cpp]: done. Out of curiosity, what's the difference ? (always though it was the same thing, but for C++)

Comment (wolfpld) [reply test/test.cpp]: `csignal` guarantees that functions are available in the `std` namespace, and *maybe* in the global one.
`signal.h` guarantees that functions are available in the global namespace, and *maybe* in `std`.

Comment (Arpafaucon) [reply test/test.cpp]: thanks for the explanation !
